libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_transport.c

```diff
@@ -844,101 +955,103 @@ static MQTTStatus_t handleIncomingPingresp( MQTTPacketInfo_t * packet,
  * @param[in] buf A pointer to a buffer containing data to be sent out.
  * @param[in] bytesToWrite number of bytes to write from the buffer.
  */
-int32_t IotBleMqttTransportSend( NetworkContext_t * pContext,
-                                 void * buf,
+int32_t IotBleMqttTransportSend( const NetworkContext_t * pContext,
+                                 const void * pBuffer,
                                  size_t bytesToWrite )
 {
-    size_t CBORbytesWritten = 0;
-    uint8_t packetType = *( ( uint8_t * ) buf );
+    size_t bytesSent = 0;
+    uint8_t packetType = ( *( ( uint8_t * ) pBuffer ) & 0xF0 );
     MQTTStatus_t status = MQTTSuccess;
-    MQTTFixedBuffer_t bufToSend = { 0 };
-    size_t packetSize = 0;
+    MQTTFixedBuffer_t serializedBuf = { 0 };
 
     /* The send function returns the CBOR bytes written, so need to return 0 or full amount of bytes sent. */
-    int32_t MQTTBytesWritten = ( int32_t ) bytesToWrite;
+    int32_t bytesWritten = ( int32_t ) bytesToWrite;
 
-    switch( packetType )
+    if( pContext->publishInfo.pending == true )
```

Comment:
A comment would be better why this check is required here. Is this to handle the payload part of the PUBLISH?

Reply:
Yes, its used to handle the publish information in parts from managed API. I have added a comment for this.